### PR TITLE
Structural/improving yield surfaces

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this PR I'm correcting an error in the Drucker-Prager yield surface and I've optimized some operations in the rest of yields and potentials.

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
Thanks

**Comment by loumalouomega** [comment]
Tests and compilation are failing

**Comment by loumalouomega** [comment]
Look like Travis is falling

**Comment by loumalouomega** [comment]
Tests are failing, can you check why?

**Comment by loumalouomega** [comment]
Still one fail:

~~~
Ran 683 of 684 test cases in 8.4254s. 1 failed:
    TestConstitutiveLawIntegrateStressPlasticitySmallStrain Failed with message: 
        Error: Check failed because std::abs((DPres[comp] - TestDP[comp])/DPres[comp]) is greater than tolerance
in /home/travis/build/KratosMultiphysics/Kratos/applications/StructuralMechanicsApplication/tests/cpp_tests/test_plasticity_constitutive_laws_3d.cpp:163:virtual void Testing::TestConstitutiveLawIntegrateStressPlasticitySmallStrain::TestFunction()
~~~

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_yield_surfaces_3d.cpp [hunk 1]
```diff
@@ -146,9 +146,10 @@ KRATOS_TEST_CASE_IN_SUITE(YieldSurfacesDerivatives, KratosStructuralMechanicsFas
     DP::CalculateYieldSurfaceDerivative(Stress, Deviator, J2, TestDP, aux_param);
     T::CalculateYieldSurfaceDerivative(Stress, Deviator, J2, TestT, aux_param);
 
+    KRATOS_WATCH(TestDP)
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_yield_surfaces_3d.cpp]: Leftover

Comment (AlejandroCornejo) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_yield_surfaces_3d.cpp]: yes! thanks

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**